ui/src/pages/my-bookmarks.tsx

```diff
@@ -11,38 +11,62 @@ import * as api from '@api';
 
 export const getServerSideProps: Types.GetServerSideProps = Helpers.withServerSession(async (context) => {
     const token = Helpers.getJWT(context);
-    let usersBookmarks: Interfaces.BookmarkedPublicationsData[] = [];
+    let userPublicationBookmarks: Interfaces.BookmarkedEntityData[] = [];
+    let userTopicBookmarks: Interfaces.BookmarkedEntityData[] = [];
 
     try {
-        const response = await api.get(`${Config.endpoints.bookmarks}`, token);
-        usersBookmarks = response.data;
+        const response = await api.get(`${Config.endpoints.bookmarks}?type=PUBLICATION`, token);
+        userPublicationBookmarks = response.data;
+    } catch (err) {
+        console.log(err);
+    }
+
+    try {
+        const response = await api.get(`${Config.endpoints.bookmarks}?type=TOPIC`, token);
+        userTopicBookmarks = response.data;
     } catch (err) {
         console.log(err);
     }
 
     return {
         props: {
-            usersBookmarks,
+            userPublicationBookmarks,
+            userTopicBookmarks,
             token,
             protectedPage: true
         }
     };
 });
 
 type Props = {
-    usersBookmarks: Interfaces.BookmarkedPublicationsData[];
+    userPublicationBookmarks: Interfaces.BookmarkedEntityData[];
+    userTopicBookmarks: Interfaces.BookmarkedEntityData[];
     token: string;
 };
 
 const MyBookmarks: Types.NextPage<Props> = (props): React.ReactElement => {
-    const [userBookmarks, setUserBookmarks] = React.useState(props.usersBookmarks);
+    const [userPublicationBookmarks, setUserPublicationBookmarks] = React.useState(props.userPublicationBookmarks);
+    const [userTopicBookmarks, setUserTopicBookmarks] = React.useState(props.userTopicBookmarks);
 
-    const deleteBookmark = async (publication: string) => {
+    const deletePublicationBookmark = async (id: string) => {
+        try {
+            await api.destroy(`${Config.endpoints.bookmarks}/${id}`, props.token);
+
+            const getPublicationBookmarks = await api.get(
```

Comment:
if the bookmark is deleted successfully, we can simply update the state without it, fetching the bookmarks again is not needed.

Reply:
Yes, corrected now